modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicStore.java

```diff
@@ -0,0 +1,484 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2022, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.vectormosaic;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.swing.event.ChangeEvent;
+import javax.swing.event.ChangeListener;
+import org.geotools.data.DataStore;
+import org.geotools.data.DataStoreFactorySpi;
+import org.geotools.data.DataStoreFinder;
+import org.geotools.data.Repository;
+import org.geotools.data.simple.SimpleFeatureCollection;
+import org.geotools.data.simple.SimpleFeatureIterator;
+import org.geotools.data.simple.SimpleFeatureSource;
+import org.geotools.data.store.ContentDataStore;
+import org.geotools.data.store.ContentEntry;
+import org.geotools.data.store.ContentFeatureSource;
+import org.geotools.feature.NameImpl;
+import org.geotools.util.SoftValueHashMap;
+import org.geotools.util.factory.GeoTools;
+import org.geotools.util.logging.Logging;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.Name;
+
+/** {@link DataStore} for a vector mosaic. */
+public class VectorMosaicStore extends ContentDataStore {
+    static final Logger LOGGER = Logging.getLogger(VectorMosaicStore.class);
+
+    @SuppressWarnings("PMD.UnusedPrivateField") // field is used in constructor
+    private final Repository repository;
+
+    DataStore delegateDataStore;
+    boolean tolerant;
+
+    public static final String PREFERRED_SPI_DEFAULT =
+            "org.geotools.data.shapefile.ShapefileDataStoreFactory";
+
+    String preferredSPI = PREFERRED_SPI_DEFAULT;
```

Comment:
IntelliJ calls out a "redundant initializer", because the sole constructor will always override this value. Meaning the Shapefile is not really used.
Make it the sample value of the factory parameter instead.

Reply:
Fixed